### DISCUSSION TITLE
How to enable XML files output in Eleventy

### DISCUSSION THREAD

**Discussion by jcubic** [11ty/eleventy#2987] [Q&A] [answered]
I've create xml file and it just copy pasted into the output. This my file:

```liquid
---
layout: null
permalink: /blog/rss.xml
eleventyExcludeFromCollections: true
---
<?xml version="1.0" encoding="UTF-8"?>
<rss version="2.0" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>Jakub Jankiewicz Blog</title>
    <link>https://jakub.jankiewicz.org/blog/</link>
    <description>Personal blog of Jakub Jankiewicz</description>
    <lastBuildDate>{{ site.date | date: "%a, %d %b %Y %H:%M:%S %Z" }}</lastBuildDate>

    {% for post in collections.posts %}
      <item>
        <title>{{ post.data.title }}</title>
        <link>{{ post.url }}</link>
        <description>{{ post.templateContent | xml_escape }}</description>
        <pubDate>{{ post.date | date: "%a, %d %b %Y %H:%M:%S %Z" }}</pubDate>
        <dc:creator>{{ metadata.author.name }}</dc:creator>
        <guid isPermaLink="true">{{ post.url }}</guid>
      </item>
    {% endfor %}
  </channel>
</rss>
```

I can't find information where I should add extensions that are processed by eleventy.

I tried what is in documentation: https://www.11ty.dev/docs/plugins/rss/
Copy-pated whole file and got the same issue, the output contain all the code instead of XML.

I've tried what chatGPT was suggesting:

```javascript
eleventyConfig.addTemplateFormats("xml");
```
but then no output file is generated.

**Comment by jcubic** [comment, marked as answer]
So I was able to solve it, I'm not sure if this is in documentation but you can't use xml extension with Eleventy like with Jekyll. You need liquid extension and use permalink to copy to the output.

The documentation in RSS say to name the file njk, but I think that this should be big warning.

**Reply by cfjedimaster** [reply]
To be clear though, I think this is clearly documented. If you want a dynamic file, you have to use one of the template languages (liquid, njk, etc), and if you don't want output to something.html, you use permalink.

**Reply by jcubic** [reply]
@cfjedimaster May I ask where it's documentation that can't have html or xml files like in Jekyll?

**Reply by cfjedimaster** [reply]
It's a template language identified by the extension. It has to be identified _somehow_, right? How is Eleventy going to know to process foo.xml as Liquid w/o the extension? Now, there's some exceptions to this - Markdown files can be processed w/ a template language too, but I honestly feel like this isn't something that necessarily needs to be called out.

**Reply by pdehaan** [reply]
And to clarify, you can use .html files as templates in your source directory https://www.11ty.dev/docs/languages/html/ and they will get preprocessed by LiquidJS.
Although there is also an `htmlTemplateEngine` config to change that if you want a different template engine: https://www.11ty.dev/docs/config/#default-template-engine-for-html-files.
